<?xml version="1.0"?>
<launch>
    <arg name="veh" default="ASV"/>
    <arg name="gazebo" default="false"/>
    <arg name="differential_constrain" default="0.2" />
    <arg name="sec" default="4"/>
    <arg name="port" default="/dev/motor"/>
    <arg name="baud" default="57600"/>
    <arg name="g29" default="false"/>

    <group ns="$(arg veh)">
        <remap from="joymapper/joy" to="joy" />
        <remap if="$(arg gazebo)" from="/mavros/global_position/raw/fix" to="/gps"/>
        <remap if="$(arg gazebo)"  from="/mavros/imu/data" to="/imu/data"/>
        <group if="$(arg gazebo)">
            <node pkg="joystick_control" type="joy_node.py" name="joymapper" output="screen">
                <param name="gazebo" value="$(arg gazebo)" />
            </node>
            <node pkg="joystick_control" type="check_signal.py" name="check_signal" output="screen">
                <param name="sec" value="$(arg sec)" />
            </node>
            <node pkg="joystick_control" type="check_internet.py" name="check_internet" output="screen">
                <param name="sec" value="$(arg sec)" />
            </node>
        </group>

        <group unless="$(arg gazebo)">
            <group if="$(arg g29)">
                <remap from="g29_joy" to="joy" />
                <node pkg="joystick_control" type="joy_g29.py" name="joy_g29" output="screen">
                </node>
                <remap from="joy" to="mapping_joy" />
                <node pkg="joystick_control" type="joy_node.py" name="joymapper" output="screen">
                    <param name="gazebo" value="$(arg gazebo)" />
                </node>
            </group>
            <group unless="$(arg g29)">
                <node pkg="joystick_control" type="joy_node.py" name="joymapper" output="screen">
                    <param name="gazebo" value="$(arg gazebo)" />
                </node>
            </group>
            <node pkg="joystick_control" type="check_signal.py" name="check_signal" output="screen">
                <param name="sec" value="$(arg sec)" />
            </node>
            <node pkg="joystick_control" type="check_internet.py" name="check_internet" output="screen">
                <param name="sec" value="$(arg sec)" />
            </node>
            <node pkg="rosserial_python" type="serial_node.py" name="rosserial" output="screen">
                <param name="port" value="$(arg port)" />
                <param name="baud" value="$(arg baud)" />
            </node>
        </group>
    </group>

</launch>
